service/auth.go

```diff
@@ -0,0 +1,67 @@
+package service
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+
+	authclient "github.com/fabric8-services/fabric8-auth-client/auth"
+	goaclient "github.com/goadesign/goa/client"
+	"github.com/goadesign/goa/middleware/security/jwt"
+	errs "github.com/pkg/errors"
+)
+
+type Auth interface {
+	CheckSpaceScope(ctx context.Context, spaceID, requiredScope string) (bool, error)
+}
+
+func NewAuthService(hostURL string) (Auth, error) {
```

Comment:
would it make sense to provide `options` to configure/change client's transport so we can use a mock library (go-vcr for example) ?

Reply:
I guess that `opt` will be `variadic arg` so no issue if it's added in future when required but as of now I am not using it.

Reply:
@alexeykazakov for first, there will be `variadic arg` and if it will be added in future than it will not break existing consumer.

for second, can you tell me the what's the problem with creating http_client in constructor OR what's the benefit of taking everything with `CheckResourceScope()` ??  There are diff ways and if there is no major problem then what's the motivation for author to change already done stuff ??

Reply:
Right. Variadic functions will work fine.

It’s just an old pattern to not add additional work to constructors when it’s not really necessary. But maybe it’s not really relevant here.

Reply:
really wondering, if this is such a imp thing that it has to has `option` param .. !!!
I will add it.

Reply:
it seems this is minor and can be handle later if req, so I will not add `option` param.